packages/vaex-ml/vaex/ml/sklearn.py

```diff
@@ -95,3 +95,141 @@ def fit(self, df, target, **kwargs):
+
+    Note that the `.fit` method will use as much memory as needed to copy one
+    batch of data, while the `.predict` method makes a full memory copy.
+    The `transform` method is evaluated lazily, and no memory copies are made.
+
+    The scikit-learn estimators themselves are not modified at all, they are
+    taken from your local installation of scikit-learn.
+
+    Example:
+
+    >>> import vaex
+    >>> import vaex.ml
+    >>> from vaex.ml.sklearn import IncrementalPredictor
+    >>> from sklearn.linear_model import SGDRegressor
+    >>>
+    >>> df = vaex.example()
+    >>>
+    >>> features = df.column_names[:6]
+    >>> target = 'FeH'
+    >>>
+    >>> standard_scaler = vaex.ml.StandardScaler(features=features)
+    >>> df = standard_scaler.fit_transform(df)
+    >>>
+    >>> features = df.get_column_names(regex='^standard')
+    >>> model = SGDRegressor(learning_rate='constant', eta0=0.01, random_state=42)
+    >>>
+    >>> incremental = IncrementalPredictor(model=model,
+    ...                                    features=features,
+    ...                                    batch_size=10_000,
+    ...                                    num_epochs=3,
+    ...                                    shuffle=True,
+    ...                                    prediction_name='pred_FeH')
+    >>> incremental.fit(df=df, target=target)
+    >>> df = incremental.transform(df)
+    >>> df.head(5)[['FeH', 'pred_FeH']]
+      #        FeH    pred_FeH
+      0  -2.30923     -1.66226
+      1  -1.78874     -1.68218
+      2  -0.761811    -1.59562
+      3  -1.52088     -1.62225
+      4  -2.65534     -1.61991
+    '''
+
+    model = traitlets.Any(default_value=None, allow_none=True, help='A scikit-learn estimator with `.fit_predict` method.').tag(**serialize_pickle)
+    features = traitlets.List(traitlets.Unicode(), help='List of features to use.')
+    batch_size = traitlets.Int(default_value=1_000_000, allow_none=False, help='Number of samples to be sent to the model in each batch.')
+    num_epochs = traitlets.Int(default_value=1, allow_none=False, help='Number of times each batch is sent to the model.')
+    shuffle = traitlets.Bool(default_value=False, allow_none=False, help='If True, shuffle the samples before sending them to the model.')
+    prediction_name = traitlets.Unicode(default_value='prediction', help='The name of the virtual column housing the predictions.')
+
+    def __call__(self, *args):
+        X = np.vstack([arg.astype(np.float64) for arg in args]).T.copy()
+        return self.model.predict(X)
+
+    def predict(self, df):
+        '''Get an in-memory numpy array with the predictions of the SKLearnPredictor.self
+
+        :param df: A vaex DataFrame, containing the input features.
+        :returns: A in-memory numpy array containing the SKLearnPredictor predictions.
+        :rtype: numpy.array
+        '''
+        data = df[self.features].values
+        return self.model.predict(data)
```

Comment:
why don't do here
```
return self.transform(df)[self.prediction_name].values
```
should be much more efficient right? should also update the docs for that

Reply:
we could indeed!